src/syscheckd/src/registry/registry.c

```diff
@@ -160,6 +161,11 @@ static void registry_key_transaction_callback(ReturnTypeCallback resultType,
     cJSON_AddStringToObject(data, "mode", FIM_EVENT_MODE[event_data->evt_data->mode]);
     cJSON_AddStringToObject(data, "type", FIM_EVENT_TYPE_ARRAY[event_data->evt_data->type]);
     cJSON_AddStringToObject(data, "arch", arch == ARCH_32BIT ? "[x32]" : "[x64]");
+    if(timestamp = cJSON_GetObjectItem(dbsync_event, "last_event"), timestamp != NULL){
+        cJSON_AddNumberToObject(data, "timestamp", timestamp->valueint);
+    }else{
+        cJSON_AddNumberToObject(data, "timestamp", key->last_event);
+    }
```

Comment:
```suggestion
    if(timestamp = cJSON_GetObjectItem(dbsync_event, "last_event"), timestamp != NULL) {
        cJSON_AddNumberToObject(data, "timestamp", timestamp->valueint);
    } else {
        cJSON_AddNumberToObject(data, "timestamp", key->last_event);
    }
```

Reply:
Done in [commit](https://github.com/wazuh/wazuh/pull/12359/commits/224cd150ea90378c265d90b500fd7e98d723ab46)